perses/tests/test_repex.py

```diff
@@ -188,5 +189,229 @@ def test_RESTCapableHybridTopologyFactory_repex_charge_mutation():
+            fe_setup = RelativeFEPSetup(
+                ligand_input=test_file_name,
+                old_ligand_index=ligand_A_index,
+                new_ligand_index=ligand_B_index,
+                forcefield_files=["amber14/tip3p.xml"],
+                small_molecule_forcefield="gaff-2.11",
+                phases=["solvent"],
+                solvent_padding=1.1 * unit.nanometers)
+
+            # Generate htf
+            htf = RESTCapableHybridTopologyFactory(
+                topology_proposal=fe_setup.solvent_topology_proposal,
+                current_positions=fe_setup.solvent_old_positions,
+                new_positions=fe_setup.solvent_new_positions
+            )
+
+            # Make sure LRC is set correctly
+            hybrid_system = htf.hybrid_system
+            force_dict = {force.getName(): index for index, force in enumerate(hybrid_system.getForces())}
+            custom_force = hybrid_system.getForce(force_dict['CustomNonbondedForce_sterics'])
+            nonbonded_force = hybrid_system.getForce(force_dict['NonbondedForce_sterics'])
+            custom_force.setUseLongRangeCorrection(False)
+            nonbonded_force.setUseDispersionCorrection(True)
+
+            # Create a reporter
+            selection = "not water"
+            reporter_file = os.path.join(temp_dir, f"{ligand_A_index}_{ligand_B_index}.nc")
+            reporter = MultiStateReporter(
+                reporter_file,
+                analysis_particle_indices=htf.hybrid_topology.select(selection),
+                checkpoint_interval=100)
+
+            # Build the hybrid repex sampler
+            sampler = HybridRepexSampler(
+                mcmc_moves=mcmc.LangevinDynamicsMove(
+                    timestep=4.0 * unit.femtoseconds,
+                    collision_rate=1.0 / unit.picosecond,
+                    n_steps=250,
+                    reassign_velocities=False,
+                    n_restart_attempts=20,
+                    constraint_tolerance=1e-06),
+                replica_mixing_scheme='swap-all',
+                hybrid_factory=htf,
+                online_analysis_interval=None)
+
+            sampler.setup(
+                n_states=n_states,
+                temperature=300 * unit.kelvin,
+                storage_file=reporter,
+                minimisation_steps=0,
+                endstates=True)
+
+            sampler.energy_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+            sampler.sampler_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+
+            # Run repex
+            sampler.extend(n_iterations)
+
+            # Retrieve free energy
+            reporter.close()
+            reporter = MultiStateReporter(reporter_file)
+            analyzer = MultiStateSamplerAnalyzer(reporter, max_n_iterations=n_iterations)
+            f_ij, df_ij = analyzer.get_free_energy()
```

Comment:
Ah I don't know why I didn't think of this before -- we may want to try bootstrapping the uncertainty. This will make the uncertainty larger (otherwise, MBAR underestimates the uncertainty) which may make the test pass!

Reply:
Yes, I think we had the same question with @dominicrufa . This may be needed not only for the tests but in general for analysis, I think.